task01/src/Vector.cpp

```diff
@@ -0,0 +1,318 @@
+#pragma once
+#include "Vector.h"
+
+
+//--------------------------------------------------------------------------------
+
+/*MOVE constructor*/
+Vector::Vector(Vector&& other) noexcept : m_size(other.m_size), m_capacity(other.m_size), m_array(other.m_array) {
+    /*std::cout << "moved" << std::endl;*/
+
+    other.m_size = 0;
+    other.m_capacity = 0;
+    other.m_array = nullptr;
+
+}
+//--------------------------------------------------------------------------------
+/*Vector class constructor*/
+Vector::Vector(std::size_t size, int value)
+{
+    // Set the member variable m_size to the value of size
+    m_size = size;
+    
+    // Set the member variable m_capacity to the value of size
+    m_capacity = size;
+    
+    // Allocate memory for the array using the new operator
+    m_array = new int[m_capacity];
+    
+    // Initialize each element of the array with the value provided
+    for (int i = 0; i < m_size; i++)
+    {
+        m_array[i] = value;
+    }
+
+}
+
+//--------------------------------------------------------------------------------
+// Copy constructor
+Vector::Vector(const Vector& other)
+{
+    // Copy the size and capacity of the other vector to the lvalue vector's member variables
+    m_size = other.m_size;
+    m_capacity = other.m_capacity;
+
+    // Allocate a new array for the lvalue vector using the capacity of the other vector
+    m_array = new int[m_capacity];
+
+    // Copy the elements of the other vector's array to the lvalue vector's array
+    for (std::size_t i = 0; i < m_size; ++i) {
+        m_array[i] = other.m_array[i];
+    }
+
+}
+//--------------------------------------------------------------------------------
+/*copy assignment operator*/
+Vector& Vector::operator=(const Vector& other)
```

Comment:
This method can be implemented in a much easier way.
Check out the "copy-and-swap idiom"